yt/frontends/enzo_e/fields.py

```diff
@@ -128,13 +128,13 @@ def setup_particle_fields(self, ptype, ftype="gas", num_neighbors=64):
         self.setup_particle_mass_field(ptype)
 
     def setup_particle_mass_field(self, ptype):
-        name = "particle_mass"
+        fname = "particle_mass"
         if ptype in self.ds.particle_unions:
-            add_union_field(self, ptype, name, "code_mass")
+            add_union_field(self, ptype, fname, "code_mass")
             return
```

Comment:
Can you clarify what is going on here? i.e., what particle types are in `self.ds.particle_unions`? And what does `add_union_fields` do here?

Reply:
Sure, `setup_particle_mass_field` is going to get called for particle types that represent the combination of multiple on-disk particle types, namely the "all" particle type which will represent all particles in the dataset. `add_union_field` is responsible for adding fields represents by these combined types. The `name` to `fname` change was only because I didn't like that I was using both `name` and `names` is rather different contexts.

Reply:
To be honest, I'm still a little confused. For a concrete example, if we have simulation data with both "dark" and "star" particles, will yt also define an "all" particle type, which will include both "dark" and "star" particles? In this case, will `self.ds.particle_unions` just contain the "all" particle type? If so, does that mean that if `setup_particle_mass_field` is called for the "all" particle type, the `return` statement on line 134 is executed and no conversion from "density" to "mass" happens?

Reply:
`all` will include all particle fields, yes.  The fields associated with particle unions are all of the fields that can be generated for all of the particle fields in that union.  (I suppose it's moderately humorous to note that the fields accessible for a "particle union" are in fact the intersection of the fields for the components.)

Reply:
Ah, the return is there because we don't need to define the field functions for the union types. We only need add the field representing the union and it will call the field functions for all the encompassing types.